### PR TITLE
Task - Setting up Database and Table

### PR DESCRIPTION
**Pull request by Huulamnguyen**
**What does this PR do?**
- This PR sets up a database and tables

**Description of Task to be completed?**
- Set up table users and table favorite

**How should this be manually tested?**
NAN

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    @classmethod
+    def signup(cls, username, email, password, image):
+        """ Sign up user.
+            Hashed password and add user to the system """
+        
+        hashed_pwd = bcrypt.generate_password_hash(password).decode('UTF-8')
+        user = User(username=username, email=email, password=password, image=image)
+        db.session.add(user)
+        return user
```
[the remark this answers is not in the document]
Comment (Huulamnguyen) [reply models.py]: Hello @alexkayabula, currently at this point, this @classmethod doesn’t relate at all to **setting-up databases and tables**. However, I believe in the future, I will add it so that I decided to add it at the beginning. If it isn’t right, I will remove it and add it later when I work on the sign-up/register feature.

File: models.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        """ Validate that user exists & password is correct.
+            Return user if valid; else return False."""
+
+        user = cls.query.filter_by(username=username).first()
+        if user:
+            is_auth = bcrypt.check_password_hash(user.password, password)
+            if is_auth:
+                return user
+        
+        return False
```
[the remark this answers is not in the document]
Comment (Huulamnguyen) [reply models.py:55]: Hello @alexkayabula , the same answer as above

File: models.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    @classmethod
+    def signup(cls, username, email, password, image_url):
+        """ Sign up user.
+            Hashed password and add user to the system """
+        
+        hashed_pwd = bcrypt.generate_password_hash(password).decode('UTF-8')
+        user = User(username=username, email=email, password=password, image_url=image_url)
+        db.session.add(user)
+        return user
```
Comment (alexkayabula) [inline models.py:42]: @Huulamnguyen How is this related to this pull request ?

File: models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexkayabula) [reply models.py]: Ok then, please refactor accordingly and change the label.

File: models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexkayabula) [reply models.py:55]: Ok then, please refactor accordingly and change the label.

[the remark this answers is not in the document]
Comment (Huulamnguyen) [reply models.py]: @alexkayabula Please review it again

**Comment by alexkayabula** [comment]
Thank you.LGTM
